src/app/app.component.ts

```diff
@@ -7,4 +8,15 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'splitwise';
+  showLoginButton = true;
+  constructor(private router: Router) {
+    this.router.events.subscribe((val) => {
```

Comment:
The default value initialization of any variable will be inside the constructor.
showLoginButton = false; or showLoginButton = true;